vtr_flow/scripts/python_libs/vtr/task.py

```diff
@@ -0,0 +1,625 @@
+
+class Job:
+    """
+    A class to store the nessesary information for a job that needs to be run.
+    """
+
+    def __init__(
+        self,
+        task_name,
+        arch,
+        circuit,
+        script_params,
+        work_dir,
+        run_command,
+        parse_command,
+        second_parse_command,
+        qor_parse_command,
+    ):
+        self._task_name = task_name
+        self._arch = arch
+        self._circuit = circuit
+        self._script_params = script_params
+        self._run_command = run_command
+        self._parse_command = parse_command
+        self._second_parse_command = second_parse_command
+        self._qor_parse_command = qor_parse_command
+        self._work_dir = work_dir
+
+    def task_name(self):
+        """
+        return the task name of the job
+        """
+        return self._task_name
+
+    def arch(self):
+        """
+        return the architecture file name of the job
+        """
+        return self._arch
+
+    def circuit(self):
+        """
+        return the circuit file name of the job
+        """
+        return self._circuit
+
+    def script_params(self):
+        """
+        return the script parameter of the job
+        """
+        return self._script_params
+
+    def job_name(self):
+        """
+        return the name of the job
+        """
+        return str(PurePath(self.arch()) / self.circuit() / self.script_params())
+
+    def run_command(self):
+        """
+        return the run command of the job
+        """
+        return self._run_command
+
+    def parse_command(self):
+        """
+        return the parse command of the job
+        """
+        return self._parse_command
+
+    def second_parse_command(self):
+        """
+        return the second parse command of the job
+        """
+        return self._second_parse_command
+
+    def qor_parse_command(self):
+        """
+        return the qor parse command of the job
+        """
+        return self._qor_parse_command
+
+    def work_dir(self, run_dir):
+        """
+        return the work directory of the job
+        """
+        return str(PurePath(run_dir).joinpath(self._work_dir))
+
+# pylint: enable=too-many-instance-attributes
+
+def load_task_config(config_file):
+    """
+    Load task config information
+    """
+    # Load the file stripping comments
+    values = load_list_file(config_file)
+
+    # Keys that can appear only once
+    unique_keys = set(
+        [
+            "circuits_dir",
+            "archs_dir",
+            "parse_file",
+            "script_path",
+            "script_params",
+            "script_params_common",
+            "pass_requirements_file",
+            "sdc_dir",
+            "qor_parse_file",
+            "cmos_tech_behavior",
+            "pad_file",
+        ]
+    )
+
+    # Keys that are repeated to create lists
+    repeated_keys = set(["circuit_list_add", "arch_list_add", "script_params_list_add"])
```

Comment:
The pattern is that lists are built from *_list_add parameters.

Reply:
This has been fixed